zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -450,11 +450,11 @@ where
         // Prevent loss of miner funds due to an unsupported or incorrect address type.
         if let Some(miner_address) = mining_config.miner_address.clone() {
             assert_eq!(
-                miner_address.network(),
-                network.clone(),
+                miner_address.network_kind(),
+                network.kind(),
                 "incorrect miner address config: {miner_address} \
                          network.network {network} and miner address network {} must match",
```

Comment:
###### Optional Suggestion

```suggestion
                "Incorrect mining address config: Zebra is configured to run on {}, but the \
                         configured mining address is for {}. The network kinds must match.",
                network.kind(),
```

Reply:
Oh. This could cause problems on Regtest, since the miner address will always be `NetworkKind::Testnet`.

Reply:
Fixed in https://github.com/ZcashFoundation/zebra/pull/8383/commits/325a6c615491d79fdc699f7bdf3cddbcb9c06a71.